Mathlib/Order/CompleteLattice/MulticoequalizerDiagram.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2025 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.Order.CompleteLattice.Lemmas
+import Mathlib.CategoryTheory.Category.Preorder
+import Mathlib.CategoryTheory.Limits.Shapes.Multiequalizer
+import Mathlib.CategoryTheory.CommSq
+import Mathlib.Tactic.FinCases
+
+/-!
+# Multicoequalizer diagrams in complete lattices
+
+We introduce the notion of bicartesian square (`Lattice.BicartSq`) in a lattice `T`.
+This consists of elements `x₁`, `x₂`, `x₃` and `x₄` such that `x₂ ⊔ x₃ = x₄` and
+`x₂ ⊓ x₃ = x₁`.
+
+It shall be shown (TODO) that if `T := Set X`, then the image of the
+associated commutative square in the category `Type _` is a bicartesian square
+in a categorical sense (both pushout and pullback).
+
+More generally, if `T` is a complete lattice, `x : T`, `u : ι → T`, `v : ι → ι → T`,
+we introduce a property `MulticoequalizerDiagram x u v` which says that `x` is
+the supremum of `u`, and that for all `i` and `j`, `v i j` is the minimum of `u i` and `u j`.
+Again, when `T := Set X`, we shall show (TOOD) that we obtain a multicoequalizer diagram
+in the category of types.
+
+-/
+
+universe u
+
+open CategoryTheory Limits
+
```

Comment:
Note that we can use the ✨Shiny New Thing✨ to clear some of the proofs, at the cost of some local annotations

```suggestion

attribute [local grind] inf_le_left inf_le_right le_sup_left le_sup_right 

```